### ISSUE TITLE
How to add a L1 norm as regularizer 

### ISSUE THREAD

**Issue by hugozanini** [junyanz/pytorch-CycleGAN-and-pix2pix#124]
Hi,
I would like to train  network to classification and I'm I need add a L1 norm as regularizer for create a sparsity condition, but  I can't do this. I tryed construct a L1 norm by myself, but didn't work too.

I need to add l1 norm in this part of code, at the end of encoder:

`upconv = nn.ConvTranspose2d(inner_nc, outer_nc,
                                    kernel_size=4, stride=2,
                                    padding=1, bias=use_bias)`
        `down = [downrelu, downconv]`
        `up = [uprelu, upconv, upnorm]`
        `model = down + up`

How can I do this?

Thanks

**Comment by junyanz** [comment]
You probably need to output the intermediate layer's activation, and then add L1 norm on that. So you should modify your code to have `y, middle_layer=G(x)`
